app/src/main/java/org/mozilla/fenix/search/SearchController.kt

```diff
@@ -69,6 +70,12 @@ class DefaultSearchController(
         store.dispatch(SearchFragmentAction.ShowSearchShortcutEnginePicker(
             text.isEmpty() && context.settings().shouldShowSearchShortcuts
         ))
+        store.dispatch(SearchFragmentAction.ShowSearchSuggestionsHint(
+            text.isNotEmpty() &&
+                    !context.settings().shouldShowSearchSuggestionsInPrivate &&
+                    (context.asActivity() as HomeActivity).browsingModeManager.mode.isPrivate &&
```

Comment:
Instead of casting context, it would be better to have `browsingModeManager` be passed in the constructor. We should also have this be the second condition instead of the third, since it just checks the memory value instead of possibly touching the disk.

Reply:
Actually, I removed `.asActivity() ` because it wasn't needed. The passed context is always HomeActivity, so IMO it is better to keep this single reference to it in the constructor. I have moved the condition up, thank you for the suggestion, I overlooked the issue.

Reply:
In that case, could you change the `context` parameter from `Context` to `HomeActivity` to make the dependency more explicit?

```kotlin
class DefaultSearchController(
    private val activity: HomeActivity,
    private val store: SearchFragmentStore,
    private val navController: NavController
) : SearchController {
  ...
```